### PR TITLE
[IMP] hr_holidays: Allow to delete non validated future time off by simple user

### PR DESCRIPTION
**Pull request by sofiagvaladze**
Add some tests

Task - 2428789

### CODE REVIEW

File: addons/hr_holidays/models/hr_leave.py [hunk 1]
```diff
@@ -820,10 +820,13 @@ def write(self, values):
     def _unlink_if_correct_states(self):
         error_message = _('You cannot delete a time off which is in %s state')
         state_description_values = {elem[0]: elem[1] for elem in self._fields['state']._description_selection(self.env)}
+        now = fields.Datetime.now()
 
         if not self.user_has_groups('hr_holidays.group_hr_holidays_user'):
-            if any(hol.state != 'draft' for hol in self):
+            if any(hol.state != 'draft' and hol.state != 'confirm' for hol in self):
```
Comment (kba-odoo) [inline addons/hr_holidays/models/hr_leave.py:826]:
```suggestion
            if any(hol.state not in ['draft', 'confirm'] for hol in self):
```

File: addons/hr_holidays/models/hr_leave.py [hunk 2]
```diff
@@ -820,10 +820,13 @@ def write(self, values):
...
                 raise UserError(error_message % state_description_values.get(self[:1].state))
+            if any(hol.date_from < now for hol in self):
+                raise UserError('You cannot delete a time off which is in the past')
```
Comment (kba-odoo) [inline addons/hr_holidays/models/hr_leave.py:829]: You should use the `_()` function to have the string translated

File: addons/hr_holidays/tests/test_access_rights.py [hunk 3]
```diff
@@ -715,6 +717,56 @@ def test_leave_double_validate_holiday_manager(self):
...
+    def test_leave_unlink_confirm_in_past_by_user(self):
+        """ A simple user cannot delete its leave in the past"""
+        values = {
+            'name': 'Random Leave',
+            'employee_id': self.employee_emp.id,
+            'holiday_status_id': self.leave_type.id,
+            'state': 'confirm',
+        }
+        leave = self.request_leave(self.user_employee_id, datetime.now() + relativedelta(days=-4), 1, values)
+        with self.assertRaises(UserError):
```
Comment (kba-odoo) [inline addons/hr_holidays/tests/test_access_rights.py:756]:
```suggestion
        with self.assertRaises(UserError), self.cr.savepoint():
```

File: addons/hr_holidays/tests/test_access_rights.py [hunk 4]
```diff
@@ -715,6 +717,56 @@ def test_leave_double_validate_holiday_manager(self):
...
+    def test_leave_unlink_validate_by_user(self):
+        """ A simple user cannot delete its leave in validate state"""
+        values = {
+            'name': 'Random Leave',
+            'employee_id': self.employee_emp.id,
+            'holiday_status_id': self.leave_type.id,
+        }
+        leave = self.request_leave(self.user_employee_id, datetime.now() + relativedelta(days=6), 1, values)
+        leave.with_user(self.user_hrmanager_id).write({'state': 'validate'})
+        with self.assertRaises(UserError):
```
Comment (kba-odoo) [inline addons/hr_holidays/tests/test_access_rights.py:768]:
```suggestion
        with self.assertRaises(UserError), self.cr.savepoint():
```

**Review by kbapt — CHANGES_REQUESTED** [review]
A few minor suggestions :-)

Thanks!

**Comment by tivisse** [comment]
@robodoo r+

**Pull request closed without merging**